csharp/src/Telemetry/Traces/Exporters/ExportersBuilder.cs

```diff
@@ -0,0 +1,207 @@
+            }
+
+            tracerProvider = factory.Invoke(_sourceName, _sourceVersion);
+            if (tracerProvider != null)
+            {
+                exporterName = exporterOption;
+            }
+            return tracerProvider;
+        }
+
+        public static TracerProvider NewAdbcFileTracerProvider(string sourceName, string? sourceVersion) =>
+            Sdk.CreateTracerProviderBuilder()
+                .AddSource(sourceName)
+                .ConfigureResource(resource =>
+                    resource.AddService(
+                        serviceName: sourceName,
+                        serviceVersion: sourceVersion))
+                .AddAdbcFileExporter(sourceName)
+                .Build();
+
+        public static TracerProvider NewConsoleTracerProvider(string sourceName, string? sourceVersion) =>
+            Sdk.CreateTracerProviderBuilder()
+                .AddSource(sourceName)
+                .ConfigureResource(resource =>
+                    resource.AddService(
+                        serviceName: sourceName,
+                        serviceVersion: sourceVersion))
+                .AddConsoleExporter()
+                .Build();
+
+        public static TracerProvider NewOtlpTracerProvider(string sourceName, string? sourceVersion) =>
+            Sdk.CreateTracerProviderBuilder()
+                .AddSource(sourceName)
+                .ConfigureResource(resource =>
+                    resource.AddService(
+                        serviceName: sourceName,
+                        serviceVersion: sourceVersion))
+                .AddOtlpExporter()
+                .Build();
+
+        public static TracerProvider? NewNoopTracerProvider(string sourceName, string? sourceVersion) =>
+            null;
+
+        public class Builder
+        {
+            private readonly string _sourceName;
+            private readonly string? _sourceVersion;
+            private readonly Dictionary<string, Func<string, string?, TracerProvider?>> _tracerProviderFactories;
+
+            internal string SourceName => _sourceName;
+
+            internal string? SourceVersion => _sourceVersion;
+
+            internal IReadOnlyDictionary<string, Func<string, string?, TracerProvider?>> TracerProviderFactories => _tracerProviderFactories;
+
+            public Builder(string sourceName, string? sourceVersion)
+            {
+                _sourceName = sourceName;
+                _sourceVersion = sourceVersion;
+                _tracerProviderFactories = [];
+            }
+
+            public ExportersBuilder Build()
+            {
+                if (_tracerProviderFactories.Count == 0)
+                {
+                    throw new InvalidOperationException("No options provided. Please add one or more exporter.");
+                }
+                return new ExportersBuilder(this);
+            }
+
+            public Builder AddDefaultExporters()
```

Comment:
I think this should be a flag to the builder's constructor because it's ambiguous what the behavior should be if an exporter with one of the default names is already used. In this impl it'll overwrite it with the default.

Reply:
Added an option to add the default exporters as the Builder's constructor option.